### ISSUE TITLE
Define an order to set any variables (scss or custom properties) first.

### ISSUE THREAD

**Issue by Scipionh** [hudochenkov/stylelint-order#168]
I want to define an order to set all the variables first. So here is what I did:

`'order/order': ['custom-properties', 'dollar-variables', 'declarations']`

It works, but in that order, dollar-variables will always come after custom-properties, I don't want that to be necessary the case. I want to be able to mix `custom-properties` and `dollar-variables` together as long as they come first.

Is there any existing way to do that ?

**Comment by hudochenkov** [comment]
No, there is no existing way to do this. I'm afraid adding this would be against this plugin philosophy to keep everything in a specific order.

**Comment by geoffreycrofte** [comment]
@Scipionh I do like the idea, I would love to have this kind of feature for more flexibility in all that order.

@hudochenkov I don't understand your answer, why is there a "order: flexible" option then?
If you don't have the time, what about a bit of help to add this option?

Thanks for your flexibility ;)

**Comment by hudochenkov** [comment]
> why is there a "order: flexible" option then?

You got me :) It's a legacy option from old built-in rule. And it works only for linting. `--fix` might change order to the one defined in config ignoring “flexibility”. I would like to remove flexible properties group at all. They doesn't make sense to me.

I'm not convinced yet that complicating in configuration of `order` rule and associated changes to linting logic is worth it.
